src/H5FDmpi.c

```diff
@@ -41,26 +41,42 @@
  * Programmer:	Quincey Koziol
  *              Friday, January 30, 2004
  *
+ * Changes:     Reworked function to use the ctl callback so we can get
+ *              rid of H5FD_class_mpi_t.  Since there are no real limits
+ *              on what the ctl callback can do, its file parameter can't
+ *              be constant.  Thus, I had to remove the const qualifier
+ *              on this functions file parameter as well.  Note also the
+ *              circumlocution required to use the ctl callbacks output
+ *              parameter to pass back the rank without introducing
+ *              compiler warnings.
+ *                                             JRM -- 8/13/21
```

Comment:
While I'm okay with leaving the "changes" sections in the cache code, I'd like to have that not spread to the rest of the library. We've been removing the ones that are noise and changing the useful bits to NOTE sections.

Reply:
We need a method of recording significant changes to functions and the reasons for them, so as to avoid re-inserting bugs and alert programmers who are familiar with the code in question to changes.
Can you point to a better solution to this problem?